### ISSUE TITLE
Exception: "concat" mode can only merge layers with matching output shapes

### ISSUE THREAD

**Issue by sansinghsanjay** [raghakot/ultrasound-nerve-segmentation#1]
I am getting following exception at line 61 in model.py :

Exception: "concat" mode can only merge layers with matching output shapes except for the concat axis. Layer shapes: [(None, 2, 20, 512), (None, 1, 20, 256)]

I am new to keras.

Thanks...

**Comment by raghakot** [comment]
i did not test this against theano backend. Are u getting this with tensorflow backend?

**Comment by sansinghsanjay** [comment]
I am using Python 2.7 with Tensorflow backend. Can you please specify the version of tools you are using.

I resolved that error by modifying following line (model.py):
up6 = merge([UpSampling2D()(conv5), conv4], mode='concat', concat_axis=1)
to:
up6 = merge([(conv5), conv4], mode='concat', concat_axis=3)
Similarly for up7, up8 and up9.

I am not sure that doing this is logically correct or not but now at least I am not getting any error.

Thanks & Regards,
-Sanjay

**Comment by sansinghsanjay** [comment]
I tried your updated code and I think that I was missing following instructions specified in readme file:
{
    "image_dim_ordering": "th",
    "epsilon": 1e-07,
    "floatx": "float32",
    "backend": "tensorflow"
}

I followed above instructions and changed "image_dim_ordering" from "tf" to "th". It helps and resolves this issue.

Thanks...
